### DISCUSSION TITLE
Has anyone had network DNS problems with 1.9.0

### DISCUSSION THREAD

**Discussion by alexpdp7** [siderolabs/talos#10050] [Q&A] [answered]
I just updated my test cluster and pods seem to have odd behavior; DNS is resolving hostnames to the IP of my node?

(About to leave so I cannot do troubleshooting now- just posting in case this is a known issue, although I didn't find anything...)

**Comment by smira** [comment, marked as answer]
I wonder if it's https://github.com/siderolabs/talos/pull/10031, if you have a FQDN as a hostname?

But you should provide details to understand what is wrong.

**Reply by alexpdp7** [reply]
This might be the case, `machine.network.hostname` is a FQDN. I learned about `talosctl rollback` and that has resolved the problem.

I'm happy to reupgrade and test #10031 if that is going to be useful, although for me it's easiest to just wait until 1.9.1 is out.

**Reply by alexpdp7** [reply]
Ah, I hadn't noticed 1.9.1 was out. I updated with that, and the problem went away. Thanks!
